File: backend/src/main/kotlin/com/piotrwalkusz/ium/backend/model/Product.kt
package com.piotrwalkusz.ium.backend.model

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import java.math.BigDecimal

data class Product(
        @Id
        val id: String,
        val name: String,
        val manufacturer: String,
        val price: BigDecimal,
        val quantity: Int
)

File: frontend/lib/model/product.dart
class Product {
  final String id;
  String name;
  String manufacturer;
  double price;
  int quantity;

  Product({this.id, this.name, this.manufacturer, this.price, this.quantity});

  Product.copy(Product product)
      : this.id = product.id,
        this.name = product.name,
        this.manufacturer = product.manufacturer,
        this.price = product.price,
        this.quantity = product.quantity;

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as String,
      name: json['name'] as String,
      manufacturer: json['manufacturer'] as String,
      price: json['price'].toDouble(),
      quantity: json['quantity'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'manufacturer': manufacturer,
      'price': price,
      'quantity': quantity
    };
  }
}
